### PR TITLE
Add support for Snap build

### PR DESCRIPTION
**Pull request by gjolly**
Add a `snapcraft.yaml` file to configure the snapcraft build.

To build and install the snap (assuming `snapcraft` utility is installed):

```
snapcraft
sudo snap install --classic --dangerous gmailctl_$(git describe --tags --abbrev=0)-$(git rev-parse --short HEAD)_amd64.snap
```

The snap requires "classic" isolation because it's relying on config files in $HOME.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: gmailctl
```
Comment (mbrt) [inline snap/snapcraft.yaml:1]: Is it mandatory to use the snap/ directory? If not I would prefer to move this file to the hack/ directory.

**Review by mbrt — CHANGES_REQUESTED** [review]
I'm not familiar with how snap works. How do users use it? Could you update the README if this is another way to install gmailctl?

File: snap/snapcraft.yaml (hunk 1 above)
Comment (gjolly) [reply snap/snapcraft.yaml:1]: Unfortunately, I think it is. You can check this [issue](https://github.com/canonical-web-and-design/build.snapcraft.io/issues/1094), the repo was archived but I think `/snap/snapcraft.yaml`, `/snapcraft.yaml` or `/.snapcraft.yaml` are still the only options.

**Comment by gjolly** [comment]
I just pushed an other commit to add the documentation to the README and use strict confinement. I tried to document `snapcraft.yaml` as much as I could so anyone could understand what is going on.
I also pushed it to the store if you want to try it: https://snapcraft.io/gmailctl

File: snap/snapcraft.yaml (hunk 1 above)
Comment (mbrt) [reply snap/snapcraft.yaml:1]: Gotcha, thanks.

**Review by mbrt — CHANGES_REQUESTED** [review]
Thanks for the changes and sorry for the delay in responding! We're almost there.

File: hack/editor-wrapper.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (gjolly) [reply hack/editor-wrapper.sh]: My turn to apology for the delay :smile: . I moved it under `snap/local`. I couldn't move it under `snap` directly, it was raising this error:

```
The 'snap' directory is meant specifically for snapcraft, but it contains the following non-snapcraft-related paths, which is unsupported and will cause unexpected behavior:
- editor-wrapper.sh

If you must store these files within the 'snap' directory, move them to 'snap/local', which is ignored by snapcraft.
```

**Review by mbrt — APPROVED** [review]

**Comment by mbrt** [comment]
Thank you @gjolly!

**Pull request merged**